apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -460,6 +460,10 @@ type Storage struct {
 	// Whether this disk represents rotational storage
 	Rotational bool `json:"rotational,omitempty"`
 
+	// Device type, one of: HDD, SSD, NVME
+	// +kubebuilder:validation:Enum=HDD;SSD;NVME
+	Type string `json:"type,omitempty"`
```

Comment:
Should we use a go type for the values so we can define constants?

Reply:
Yea we could, but other fields in here are strings, so I was thinking to be as similar as possible.

We could do type alias for string and use constants for it.

Main concern is, keep resulting JSON similar to one that current PR generates

Reply:
Ah, yeah, good point. Those other values come directly out of Ironic and aren't modified. These are values we're defining ourselves. So, I could go either way.

Reply:
You went to the trouble of defining a DiskType, so you might as well use it. Then you wouldn't have to cast back to string all over the place.

Reply:
Well, this is so obvious, I don't have an idea why I missed that, thanks @zaneb 
Fixing in next rebase.